### PR TITLE
Reactivate tenure scan test

### PR DESCRIPTION
**Pull request by adamtry**
Context: These tests were temporarily deactivated because they were failing on CircleCI but passing locally, and we wanted to test the functionality in development.

- Reactivate a single end-to-end test for the tenure scan, which combines the features of both previous data tests (Ensures correct data is returned, and pagination works to fetch the data)
- Remove end-to-end test for a case where there's no data in the tenure table as this isn't something we need to functionally support
- Parallelise CircleCI pipeline. This was done to speed up debugging an issue with this test failing in CI but passing locally, but  it would be nice to keep as it cuts down the CI workflow on feature branches from 5-6 minutes to 2 minutes, while still testing the same things.

Pipeline before:
(build-and-test included two separate sonarcloud scans)
image

Pipeline after:
(sonarcloud scan was extracted from build-and-test into a single parallel job)
image

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -233,24 +237,75 @@ jobs:
...
+              ignore: release, master
+      - terraform-init-and-plan-development:
+          requires:
+            - assume-role-development
+      - terraform-compliance-development:
+          requires:
+            - terraform-init-and-plan-development
+
+
   check-and-deploy-development:
```
Comment (adamtry) [inline .circleci/config.yml:271]: I made this only run on development. It's clearer and also lets the check-on-feature-branch be separate

**Review by Duslerke — APPROVED** [review]

File: ContactDetailsApi.Tests/Dockerfile [hunk 2]
```diff
@@ -36,4 +26,3 @@ RUN dotnet build -c Release -o out ContactDetailsApi/ContactDetailsApi.csproj
 RUN dotnet build -c debug -o out ContactDetailsApi.Tests/ContactDetailsApi.Tests.csproj
 
 CMD dotnet test
-RUN dotnet sonarscanner end /d:sonar.login="${SONAR_TOKEN}"
```
Comment (adamtry) [inline ContactDetailsApi.Tests/Dockerfile:39]: The sonar scan in this docker-compose is redundant - we have a separate job in CI dedicated to running a sonarcloud scan

File: ContactDetailsApi.Tests/V2/Gateway/TenureDbGatewayTests.cs [hunk 3]
```diff
@@ -56,89 +56,44 @@ private async Task InsertDataIntoDynamoDB(IEnumerable<TenureInformationDb> entit
...
-        private async Task DeleteDataInDynamoDB(IEnumerable<TenureInformationDb> entities)
-        {
-            foreach (var entity in entities)
-            {
-                await _dbFixture.DynamoDbContext.DeleteAsync(entity).ConfigureAwait(false);
-            }
-        }
-
         [Fact]
-        public async Task ScanTenuresReturnsEmptyListWhenNoData()
```
Comment (adamtry) [inline ContactDetailsApi.Tests/V2/Gateway/TenureDbGatewayTests.cs:68]: I felt this isn't functionality we need to test as we don't need to support the case of the tenure table having no data

File: ContactDetailsApi.Tests/V2/Gateway/TenureDbGatewayTests.cs [hunk 4]
```diff
@@ -56,89 +56,44 @@ private async Task InsertDataIntoDynamoDB(IEnumerable<TenureInformationDb> entit
...
+            secondResponse.PaginationDetails.HasNext.Should().BeFalse();
+            secondResponse.PaginationDetails.NextToken.Should().BeNull();
+
+            // Assert (1 & 2)
+            var combinedResults = firstResponse.Results.Concat(secondResponse.Results).ToList();
+            combinedResults.Should().BeEquivalentTo(tenures.Select(x => x.ToDomain()));
+            _logger.VerifyExact(LogLevel.Information, "Calling IDynamoDBContext.ScanAsync for TenureInformationDb", Times.Exactly(2));
         }
-
-        //[Fact(Skip ="")]
```
Comment (adamtry) [inline ContactDetailsApi.Tests/V2/Gateway/TenureDbGatewayTests.cs:81]: These two were uncommented and combined into one

**Pull request merged**